### PR TITLE
[refactor] house abigal builder and point classes

### PR DESCRIPTION
**Pull request by raihansyahdean**
### Main changes
- Added House Abigail builder for determining room points
- New system in GUI Inspector for builder
- Integration with StageManager for ghost roaming

### Tasks List
- [close #174 ]

### Attachments
image

### New Packages
N/A

### Resources 
N/A

### CODE REVIEW

File: Assets/Astralization/Scripts/Editor/DraggablePoint.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class DraggablePoint : PropertyAttribute { }
 
 
 #if UNITY_EDITOR
-[CustomEditor(typeof(StageController), true)]
+//CustomEditor(typeof(StageController), true)]
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Editor/DraggablePoint.cs]: delete this file

File: Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    #endregion
+
+    public void Load(Vector3 pos, string pointName, float radius)
+    {
+        SetLocalPosition(pos);
+        this.pointName = pointName;
+        this.radius = radius;
+    }
+}
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs]: newline eof

File: Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+using System;
+using UnityEngine;
+
+
+public class RoomPoint : MonoBehaviour
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs]: class docs

File: Assets/Astralization/Scripts/_Common/State.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        RemoveListeners();
+    }
+    protected virtual void AddListeners()
+    {
+    }
+
+    protected virtual void RemoveListeners()
+    {
+    }
+}
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/_Common/State.cs:26]: newline eof

File: Assets/Astralization/Scripts/_Common/StateMachine.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            _currentState.Exit();
+
+        _currentState = value;
+
+        if (_currentState != null)
+            _currentState.Enter();
+
+        _inTransition = false;
+    }
+}
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/_Common/StateMachine.cs:47]: newline eof

File: Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public string pointName;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs]: change to PascalCase?

File: Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    public float radius;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Managers/Stage System/RoomPoint.cs]: change to PascalCase?

**Review by rafiftaris — APPROVED** [review]

**Review by uciii — APPROVED** [review]
LG finally bisa ubah koordinatnya wkwkwk

**Pull request merged**